From 7207990e8e739fd4be50e60f512c079f72b871a1 Mon Sep 17 00:00:00 2001
From: Alan Tull <atull@altera.com>
Date: Wed, 17 Jul 2013 18:31:19 -0500
Subject: [PATCH] fpga2sdram bridges to device tree as per the bindings
 documentation

For testing, add some example fpga2sdram bridges.  We do not include
these bridges in the default kernel device tree as they are dependent
on the implementation of whatever is in the FPGA.
---
 arch/arm/boot/dts/socfpga_arria5.dts   |   32 ++++++++++++++++++++++++++++++++
 arch/arm/boot/dts/socfpga_cyclone5.dts |   32 ++++++++++++++++++++++++++++++++
 2 files changed, 64 insertions(+)

diff --git a/arch/arm/boot/dts/socfpga_arria5.dts b/arch/arm/boot/dts/socfpga_arria5.dts
index 18b1003..531e902 100644
--- a/arch/arm/boot/dts/socfpga_arria5.dts
+++ b/arch/arm/boot/dts/socfpga_arria5.dts
@@ -187,6 +187,38 @@
 				gpios = <&gpio0 18 1>;
 			};
 		};
+
+		hps_fpgabridge3: fpgabridge@3 {
+			compatible = "altr,socfpga-fpga2sdram-bridge";
+			label = "fpga2sdram0";
+			read-port = <0>;
+			write-port = <0>;
+			control-ports = <0 2>; /* ports 0 and 1 */
+		};
+
+		hps_fpgabridge4: fpgabridge@4 {
+			compatible = "altr,socfpga-fpga2sdram-bridge";
+			label = "fpga2sdram1";
+			read-port = <1>;
+			write-port = <1>;
+			control-ports = <2 1>; /* port 2 only */
+		};
+
+		hps_fpgabridge5: fpgabridge@5 {
+			compatible = "altr,socfpga-fpga2sdram-bridge";
+			label = "fpga2sdram2";
+			read-port = <2>;
+			write-port = <2>;
+			control-ports = <3 1>; /* port 3 only */
+		};
+
+		hps_fpgabridge6: fpgabridge@6 {
+			compatible = "altr,socfpga-fpga2sdram-bridge";
+			label = "fpga2sdram3";
+			read-port = <3>;
+			write-port = <3>;
+			control-ports = <4 2>; /* ports 4 and 5 */
+		};
 	};
 };
 
diff --git a/arch/arm/boot/dts/socfpga_cyclone5.dts b/arch/arm/boot/dts/socfpga_cyclone5.dts
index 6d90c62..a372941 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5.dts
+++ b/arch/arm/boot/dts/socfpga_cyclone5.dts
@@ -172,6 +172,38 @@
 				gpios = <&gpio1 12 1>;
 			};
 		};
+
+		hps_fpgabridge3: fpgabridge@3 {
+			compatible = "altr,socfpga-fpga2sdram-bridge";
+			label = "fpga2sdram0";
+			read-port = <0>;
+			write-port = <0>;
+			control-ports = <0 2>; /* ports 0 and 1 */
+		};
+
+		hps_fpgabridge4: fpgabridge@4 {
+			compatible = "altr,socfpga-fpga2sdram-bridge";
+			label = "fpga2sdram1";
+			read-port = <1>;
+			write-port = <1>;
+			control-ports = <2 1>; /* port 2 only */
+		};
+
+		hps_fpgabridge5: fpgabridge@5 {
+			compatible = "altr,socfpga-fpga2sdram-bridge";
+			label = "fpga2sdram2";
+			read-port = <2>;
+			write-port = <2>;
+			control-ports = <3 1>; /* port 3 only */
+		};
+
+		hps_fpgabridge6: fpgabridge@6 {
+			compatible = "altr,socfpga-fpga2sdram-bridge";
+			label = "fpga2sdram3";
+			read-port = <3>;
+			write-port = <3>;
+			control-ports = <4 2>; /* ports 4 and 5 */
+		};
 	};
 };
 
-- 
1.7.9.5

